src/java.base/share/classes/java/lang/invoke/DirectMethodHandle.java

```diff
@@ -214,7 +214,6 @@ private static LambdaForm preparedLambdaForm(MemberName m, boolean adaptToSpecia
             which = LF_INVSPECIAL_IFC;
         }
         LambdaForm lform = preparedLambdaForm(mtype, which);
-        maybeCompile(lform, m);
```

Comment:
Ok, looks like this can be dropped because we always call `compileToBytecode` in `makePreparedLambdaForm`?

Reply:
Indeed, maybeCompile is actually weaker as it does not replace interpreter vmentry